process/interceptors/multiDataInterceptor.go

```diff
@@ -152,12 +157,19 @@ func (mdi *MultiDataInterceptor) ProcessReceivedMessage(message p2p.MessageP2P,
 	for index, dataBuff := range multiDataBuff {
 		var interceptedData process.InterceptedData
 		interceptedData, err = mdi.interceptedData(dataBuff, message.Peer(), fromConnectedPeer)
```

Comment:
seems like at line 159 inside mdi.interceptedData happens all the intensive processing -> signature verifications, etc.
so checking the cache at line 167 does not save much in this case.
I would suggest moving the cache check inside mdi.InterceptedData, before interceptedData.CheckValidity
Check the same for singleDataInterceptors.